test/e2e/overrides_test.go

```diff
@@ -140,6 +143,168 @@ var _ = Describe("Overrides test", Ordered, func() {
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for cert-manager controller status to become degraded")
+			err = verifyOperatorStatusCondition(certmanageroperatorclient, []string{certManagerControllerDeploymentControllerName}, invalidOperatorStatusConditions)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Checking if the resources are not added to the cert-manager controller deployment")
+			err = verifyDeploymentResources(k8sClientSet, certmanagerControllerDeployment, res, false)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("When adding invalid cert-manager webhook override resources", func() {
+
+		It("should not add the resources to the cert-manager webhook deployment", func() {
+
+			By("Adding cert-manager webhook override resources to the certmanagaer.operator object")
+			res := v1alpha1.CertManagerResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceEphemeralStorage: k8sresource.MustParse("2Gi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceEphemeralStorage: k8sresource.MustParse("1Gi"),
+				},
+			}
+			err := addOverrideResources(certmanageroperatorclient, certmanagerWebhookDeployment, res)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for cert-manager webhook controller status to become degraded")
+			err = verifyOperatorStatusCondition(certmanageroperatorclient, []string{certManagerWebhookDeploymentControllerName}, invalidOperatorStatusConditions)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Checking if the resources are not added to the cert-manager webhook deployment")
+			err = verifyDeploymentResources(k8sClientSet, certmanagerWebhookDeployment, res, false)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("When adding invalid cert-manager cainjector override resources", func() {
+
+		It("should not add the resources to the cert-manager cainjector deployment", func() {
+
+			By("Adding cert-manager cainjector override resources to the certmanagaer.operator object")
+			res := v1alpha1.CertManagerResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceEphemeralStorage: k8sresource.MustParse("2Gi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceEphemeralStorage: k8sresource.MustParse("1Gi"),
+				},
+			}
+			err := addOverrideResources(certmanageroperatorclient, certmanagerCAinjectorDeployment, res)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for cert-manager cainjector controller status to become degraded")
+			err = verifyOperatorStatusCondition(certmanageroperatorclient, []string{certManagerCAInjectorDeploymentControllerName}, invalidOperatorStatusConditions)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Checking if the resources are not added to the cert-manager cainjector deployment")
+			err = verifyDeploymentResources(k8sClientSet, certmanagerCAinjectorDeployment, res, false)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
```

Comment:
In future we can also add test cases for the following scenarios described in: https://github.com/openshift/cert-manager-operator/pull/136#discussion_r1281566526. Okay to skip for now as `TestMergeContainerResources` from https://github.com/openshift/cert-manager-operator/blob/master/pkg/controller/deployment/deploment_helper_test.go already tests similar functionality in a unit test.